build.assets/tooling/cmd/protoc-gen-eventschema/handlerequest.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"github.com/gogo/protobuf/protoc-gen-gogo/generator"
+	gogoplugin "github.com/gogo/protobuf/protoc-gen-gogo/plugin"
+	"github.com/gogo/protobuf/vanity/command"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/build.assets/tooling/lib/eventschema"
+	"github.com/gravitational/teleport/build.assets/tooling/lib/tree"
+)
+
+func handleRequest(req *gogoplugin.CodeGeneratorRequest) error {
+	if len(req.FileToGenerate) == 0 {
+		return trace.Errorf("no input file provided")
+	}
+	if len(req.FileToGenerate) > 1 {
```

Comment:
nit: as you're referring to the same variable in both ifs I think that `switch` would be more appropriate.